cluster/manifests/kube-dns/node-local-daemonset.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: dnsmasq-node
+  namespace: kube-system
+  labels:
+    application: dnsmasq-node
+spec:
+  selector:
+    matchLabels:
+      application: dnsmasq-node
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        application: dnsmasq-node
+    spec:
+      priorityClassName: system-node-critical
+      serviceAccountName: system
+      tolerations:
+      - operator: Exists
+        effect: NoSchedule
+      - operator: Exists
+        effect: NoExecute
+      volumes:
+      - name: kube-dns-config
+        configMap:
+          name: kube-dns
+          optional: true
+      containers:
+      - name: dnsmasq
+        image: registry.opensource.zalan.do/teapot/k8s-dns-dnsmasq-nanny-amd64:1.14.10
+        securityContext:
+          privileged: true
+        livenessProbe:
+          httpGet:
+            path: /healthcheck/dnsmasq
+            port: 9054
+            scheme: HTTP
+          initialDelaySeconds: 60
+          timeoutSeconds: 5
+          successThreshold: 1
+          failureThreshold: 5
+        args:
+        - -v=2
+        - -logtostderr
+        - -configDir=/etc/k8s/dns/dnsmasq-nanny
+        - -restartDnsmasq=true
+        - --
+        - -k
+        - --cache-size=10000
+        - --no-negcache
+        - --server=/cluster.local/10.3.0.10#53
+        - --server=/in-addr.arpa/10.3.0.10#53
+        - --server=/ip6.arpa/10.3.0.10#53
```

Comment:
Should we use `10.3.0.11` to go to coredns from the beginning? Then we can safely remove the kube-dns service `10.3.0.10` after dnsmasq has been enabled.